app/scripts/lib/account-tracker.test.js

```diff
@@ -41,104 +41,98 @@ function flushPromises() {
-    blockTrackerFromHookStub.getCurrentBlock = noop;
-    blockTrackerFromHookStub.getLatestBlock = noop;
-
-    getNetworkClientByIdStub = jest.fn().mockReturnValue({
-      configuration: {
-        chainId: '0x1',
-      },
-      blockTracker: blockTrackerFromHookStub,
-      provider: providerFromHook,
-    });
+    },
+    blockTracker: blockTrackerFromHookStub,
+    provider: providerFromHook,
+  });
 
-    getNetworkIdentifierStub = jest.fn();
-
-    controllerMessenger = new ControllerMessenger();
-    controllerMessenger.registerActionHandler(
-      'AccountsController:getSelectedAccount',
-      () => ({
-        id: 'accountId',
-        address: SELECTED_ADDRESS,
-      }),
-    );
-
-    accountTracker = new AccountTracker({
-      provider,
-      blockTracker: blockTrackerStub,
-      getNetworkClientById: getNetworkClientByIdStub,
-      getNetworkIdentifier: getNetworkIdentifierStub,
-      preferencesController: {
-        store: {
-          getState: () => ({
-            useMultiAccountBalanceChecker,
-          }),
-          subscribe: noop,
-        },
-      },
-      onboardingController: {
-        store: {
-          subscribe: noop,
-          getState: () => ({
-            completedOnboarding,
-          }),
-        },
+  const controllerMessenger = new ControllerMessenger();
+  controllerMessenger.registerActionHandler(
+    'AccountsController:getSelectedAccount',
+    () => ({
+      id: 'accountId',
+      address: SELECTED_ADDRESS,
+    }),
+  );
+
+  return new AccountTracker({
+    provider,
+    blockTracker: blockTrackerStub,
+    getNetworkClientById: getNetworkClientByIdStub,
+    getNetworkIdentifier: jest.fn(),
+    preferencesController: {
+      store: {
+        getState: () => ({
+          useMultiAccountBalanceChecker,
+        }),
+        subscribe: noop,
       },
-      controllerMessenger,
-      onAccountRemoved: (callback) => {
-        accountRemovedListener = callback;
+    },
+    onboardingController: {
+      store: {
+        subscribe: noop,
+        getState: () => ({
+          completedOnboarding,
+        }),
       },
-      getCurrentChainId: () => currentChainId,
-    });
+    },
+    controllerMessenger,
+    onAccountRemoved: noop,
+    getCurrentChainId: () => currentChainId,
+    ...accountTrackerOptions,
   });
+}
 
+describe('Account Tracker', () => {
   afterEach(() => {
     jest.resetAllMocks();
   });
 
   describe('start', () => {
     it('restarts the subscription to the block tracker and update accounts', async () => {
+      const blockTrackerStub = new EventEmitter();
+      blockTrackerStub.getCurrentBlock = noop;
+      blockTrackerStub.getLatestBlock = noop;
       jest.spyOn(blockTrackerStub, 'addListener').mockImplementation();
       jest.spyOn(blockTrackerStub, 'removeListener').mockImplementation();
+      const accountTracker = buildAccountTracker({
+        blockTracker: blockTrackerStub,
+      });
       const updateAccountsSpy = jest
```

Comment:
[nit] this could be returned from the buildAccountTracker helper function to reduce duplication.

```
 const { accountTracker, updateAccountsSpy } = buildAccountTracker({
        blockTracker: blockTrackerStub,
      });
```

Reply:
But overloading the helper function can sometimes be confusing... So idk

Reply:
https://github.com/MetaMask/metamask-extension/pull/22233#discussion_r1431748203